typescript/web/src/connectors/undo-store/effects/create-label.ts

```diff
@@ -160,76 +28,52 @@ export const createCreateLabelEffect = (
     client: ApolloClient<object>;
   }
 ): Effect => ({
-  do: async () => {
+  do: async (id = uuid()) => {
     const createLabelInputs = {
+      id,
       imageId,
       labelClassId: selectedLabelClassId,
       geometry,
-      labelType,
+      type: labelType,
     };
-    const { data } = await client.mutate({
+
+    const createLabelPromise = client.mutate({
       mutation: createLabelMutation,
       variables: createLabelInputs,
-      refetchQueries: ["countLabels", "getDatasets", "countLabelsOfDataset"],
+      refetchQueries: ["countLabelsOfDataset"],
       optimisticResponse: {
-        createLabel: { id: `temp-${Date.now()}`, __typename: "Label" },
+        createLabel: { id, __typename: "Label" },
       },
       update(cache, { data: mutationPayloadData }) {
-        const id = mutationPayloadData?.createLabel?.id;
-        if (typeof id !== "string") {
+        if (typeof mutationPayloadData?.createLabel?.id !== "string") {
           return;
         }
 
-        addLabelToImageInCache(cache, { ...createLabelInputs, id });
+        addLabelToImageInCache(cache, createLabelInputs);
       },
     });
 
-    if (typeof data?.createLabel?.id !== "string") {
-      throw new Error("Couldn't get the id of the newly created label");
-    }
+    // TODO: Ideally we could select the label before awaiting for the mutation to complete
+    // because the optimistic response is run synchronously.
+    // However it makes some query fails. It would be nice to try to reverse those 2 lines
+    // and make the other "cache-first"
```

Comment:
Isn't there a way to select the label "inside" the optimistic response to ensure the queries don't fail?

Reply:
That's what I wanted to do at first. 

However, it triggers re-renders in some parts of the UI.
And those parts triggers new queries looking for the details of the selected label. However, the select label doesn't yet exist while the mutation is pending and those queries fail... (I thought the race condition would be negligible but it was happening almost every time).

We should make those queries cache-first and ensure that everything they query is already available in the cache. Then, we will be able to change that.